Realm/ObjectServerTests/SwiftFlexibleSyncServerTests.swift

```diff
@@ -1186,6 +1186,30 @@ class SwiftFlexibleSyncTests: SwiftSyncTestCase {
         XCTAssertEqual(downloadCount.value, 0)
         XCTAssertEqual(uploadCount.value, 0)
     }
+
+    func testFlexibleSyncNotEnabledError() throws {
+        let appServerId = try RealmServer.shared.retrieveAppServerId(appId)
+        let syncServerId = try RealmServer.shared.retrieveSyncServiceId(appServerId: appServerId)
+        _ = try RealmServer.shared.disableSync(appServerId: appServerId, syncServiceId: syncServerId)
+        var isEnable = try RealmServer.shared.isSyncEnabled(appServerId: appServerId, syncServiceId: syncServerId)
+        XCTAssertFalse(isEnable)
+
+        let ex = expectation(description: "Waiting for error handler to be called...")
+        ex.assertForOverFulfill = false // error handler can legally be called multiple times
+        app.syncManager.errorHandler = { @Sendable (error, _) in
+            assertSyncError(error, .clientInternalError, "Sync is not enabled for this app")
```

Comment:
This isn't an internal error. None of the existing error codes are a great match for this, so we should probably add a new one. Maybe `RLMSyncErrorServerWarning`?